test/integration/builder/transaction.js

```diff
@@ -427,7 +428,35 @@ module.exports = function(knex) {
         .spread(function (ret1, ret2) {
           expect(ret1).to.equal(ret2);
         });
-     });
+    });
+
+    it('should pass the hook context to wrapIdentifier', function () {
+      const originalWrapIdentifier = knex.client.config.wrapIdentifier;
+      const spy = sinon.spy().named('calledWithContext');
+
+      function restoreWrapIdentifier() {
+        knex.client.config.wrapIdentifier = originalWrapIdentifier;
+      }
 
+      knex.client.config.wrapIdentifier = (value, wrap, context) => {
+        spy(context);
+        return wrap(value);
+      };
+
+      return knex.transaction(function(trx) {
+        return trx
+          .select()
+          .from('accounts')
+          .hookContext({ foo: 'bar' });
```

Comment:
I'll ask some second and third opinions about this name :)

Reply:
maybe queryContext() would be a bit nicer

Reply:
I actually didn't see the guy with opinions yesterday, so didn't get one. But I kind of like `.queryContext` it is not so specific that `.hookContext` is and it hints that context is bound to single query builder.

Reply:
Agreed, will update :)

Reply:
One thing though, this PR also adds that method to the schema builder. Is it okay to call it `queryContext` there as well or should be perhaps remove it altogether?